test/unit/commands_test.rb

```diff
@@ -133,6 +133,30 @@ class Assoc < HammerCLIForeman::AddAssociatedCommand
     end
   end
 
+  context "ListSearchCommand" do
+    it "should find correct results" do
+      ResourceMocks.mock_action_calls(
+        [:hosts, :index, [{ "id" => 2, "name" => "random-host",  "ip" => "192.168.100.112", "mac" => "6e:4b:3c:2c:8a:0a" }]],
+      )
+
+      class DomainOuter < HammerCLIForeman::Command
+        resource :domains
+
+        class HostsCommand < HammerCLIForeman::AssociatedListSearchCommand
+          command_name 'hosts'
+          search_resource :hosts
+
+          output HammerCLIForeman::Host::ListCommand.output_definition
+
+          build_options
+        end
+      end
+      comm = DomainOuter::HostsCommand.new("", { :adapter => :csv, :interactive => false })
+      out, err = capture_io { comm.run(["--host-id=5"]) }
```

Comment:
`--host-id` is wrong parameter in this case (it just doesn't exist, since parent command has `domains` resource, so it should be rather `--domain-id` as you designed).